### PR TITLE
chore(docs): Add RN quick start guide

### PR DESCRIPTION
**Pull request by joebuono**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by calebpollman — APPROVED** [review]

File: examples/react-native/src/features/Authenticator/WithAuthenticator/Example.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  return (
+    <View style={style.container}>
+      <SignOutButton />
+    </View>
+  );
+}
+
+const style = StyleSheet.create({
+  container: { flex: 1, alignItems: 'center', justifyContent: 'center' },
+});
```
Comment (wlee221) [inline examples/react-native/src/features/Authenticator/WithAuthenticator/Example.tsx:28]: Is this necessary for the purpose of this docs? No opinion, just curious

Comment (joebuono) [reply examples/react-native/src/features/Authenticator/WithAuthenticator/Example.tsx:28]: Probably not necessary in this case

**Comment by calebpollman** [comment]
Anything blocking this from getting merged?

**Review by ioanabrooks — APPROVED** [review]

**Comment by joebuono** [comment]
> Anything blocking this from getting merged?

blocking

**Pull request merged**